modules/modal/react/lib/ModalContent.tsx

```diff
@@ -98,7 +98,8 @@ function getFirstElementToFocus(modalEl: HTMLElement): HTMLElement {
 }
 
 const useKeyDownListener = (handleKeydown: EventListenerOrEventListenerObject) => {
-  React.useEffect(() => {
+  // `useLayoutEffect` for automation
+  React.useLayoutEffect(() => {
```

Comment:
This was difficult to reproduce. I had to add a delay to when the event listener was added. Running from the command line saw failures sooner than running Cypress through the GUI runner. Since Travis is so slow, it happened earlier. My guess is Travis is getting slower.

`useLayoutEffect` isn't strictly necessary for production because it probably isn't possible for a user to hit a key fast enough and if they did, they could just hit the key again. Automation is fast enough though. We could either slow down the automation or not allow a frame to render without a keybinding.

`useLayoutEffect` is there to prevent frames from rendering between side effects. The documentation suggests not having any UI-blocking work here. Adding an event listener is very fast in browsers and fast-enough in IE11.

Reply:
I like this solution rather than implicitly slowing down the automation.